Hacks to fbcon that make sys_reboot() work on PS3.

---
 drivers/char/vt.c             |    6 ++++++
 drivers/video/console/fbcon.c |    7 +++++--
 drivers/video/ps3fb.c         |    9 ++++++---
 3 files changed, 17 insertions(+), 5 deletions(-)

--- linux-2.6.21.orig/drivers/char/vt.c
+++ linux-2.6.21/drivers/char/vt.c
@@ -2338,6 +2338,12 @@ struct tty_driver *console_driver;
 
 #ifdef CONFIG_VT_CONSOLE
 
+void vt_console_stop(void)
+{
+	printable = 0;
+}
+EXPORT_SYMBOL_GPL(vt_console_stop);
+
 /*
  *	Console on virtual terminal
  *
--- linux-2.6.21.orig/drivers/video/console/fbcon.c
+++ linux-2.6.21/drivers/video/console/fbcon.c
@@ -196,7 +196,7 @@ static void fbcon_redraw_move(struct vc_
 static void fbcon_modechanged(struct fb_info *info);
 static void fbcon_set_all_vcs(struct fb_info *info);
 static void fbcon_start(void);
-static void fbcon_exit(void);
+void fbcon_exit(void);
 static struct class_device *fbcon_class_device;
 
 #ifdef CONFIG_MAC
@@ -458,6 +458,8 @@ static void fbcon_del_cursor_timer(struc
 {
 	struct fbcon_ops *ops = info->fbcon_par;
 
+	printk("%s:%d:\n", __func__, __LINE__);
+
 	if (info->queue.func == fb_flashcursor &&
 	    ops->flags & FBCON_FLAGS_CURSOR_TIMER) {
 		del_timer_sync(&ops->cursor_timer);
@@ -3297,7 +3299,7 @@ static void fbcon_start(void)
 	}
 }
 
-static void fbcon_exit(void)
+void fbcon_exit(void)
 {
 	struct fb_info *info;
 	int i, j, mapped;
@@ -3349,6 +3351,7 @@ static void fbcon_exit(void)
 
 	fbcon_has_exited = 1;
 }
+EXPORT_SYMBOL_GPL(fbcon_exit);
 
 static int __init fb_console_init(void)
 {
--- linux-2.6.21.orig/drivers/video/ps3fb.c
+++ linux-2.6.21/drivers/video/ps3fb.c
@@ -1144,14 +1144,17 @@ static int ps3fb_remove(struct platform_
 	return 0;
 }
 
+extern void fbcon_exit(void);
+extern void vt_console_stop(void);
+
 static void ps3fb_shutdown(struct platform_device *dev)
 {
 	printk(" -> %s:%d\n", __func__, __LINE__);
 
-	// is this stuff ok here??
-	// just set .shutdown = ps3fb_remove???
+	vt_console_stop();
+	fbcon_exit();
 
-	ps3fb_flip_ctl(0);	/* flip off */
+	ps3fb_flip_ctl(0); /* flip off */
 	ps3fb.dinfo->irq.mask = 0;
 	ps3fb_remove(dev);
 	printk(" <- %s:%d\n", __func__, __LINE__);
